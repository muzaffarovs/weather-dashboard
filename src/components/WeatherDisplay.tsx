import React from "react";

export interface WeatherData {
  city: string;
  description: string;
  icon: string;
  temperature: number;
  feelsLike: number;
  humidity: number;
  wind: number;
}

interface WeatherDisplayProps {
  data: WeatherData;
  unit: "C" | "F";
}

export const WeatherDisplay: React.FC<WeatherDisplayProps> = ({
  data,
  unit,
}) => {
  return (
    <div className="weather-display">
      <h2 className="city-name">{data.city}</h2>
      <div className="weather-main">
        <img
          src={`https://openweathermap.org/img/wn/${data.icon}@2x.png`}
          alt={data.description}
          className="weather-icon"
        />
        <div className="weather-temp">
          <h1>
            {Math.round(data.temperature)}°{unit}
          </h1>
          <p>{data.description}</p>
        </div>
      </div>
      <div className="weather-details">
        <p>
          Feels like: {Math.round(data.feelsLike)}°{unit}
        </p>
        <p>Humidity: {data.humidity}%</p>
        <p>Wind: {Math.round(data.wind)} m/s</p>
      </div>
    </div>
  );
};
