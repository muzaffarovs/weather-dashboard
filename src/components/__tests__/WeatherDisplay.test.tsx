import { render } from "@testing-library/react";
import { WeatherDisplay, type WeatherData } from "../WeatherDisplay";

describe("WeatherDisplay", () => {
  it("renders correctly and matches snapshot", () => {
    const mockData: WeatherData = {
      city: "London",
      description: "moderate rain",
      icon: "03d",
      temperature: 21,
      feelsLike: 20,
      humidity: 60,
      wind: 5.2,
    };

    const { container } = render(<WeatherDisplay data={mockData} unit="C" />);

    expect(container).toMatchSnapshot();
  });
});
