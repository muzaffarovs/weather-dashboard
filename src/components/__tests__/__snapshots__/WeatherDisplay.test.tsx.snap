// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDisplay renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="weather-display"
  >
    <h2
      class="city-name"
    >
      London
    </h2>
    <div
      class="weather-main"
    >
      <img
        alt="moderate rain"
        class="weather-icon"
        src="https://openweathermap.org/img/wn/03d@2x.png"
      />
      <div
        class="weather-temp"
      >
        <h1>
          21
          °
          C
        </h1>
        <p>
          moderate rain
        </p>
      </div>
    </div>
    <div
      class="weather-details"
    >
      <p>
        Feels like: 
        20
        °
        C
      </p>
      <p>
        Humidity: 
        60
        %
      </p>
      <p>
        Wind: 
        5
         m/s
      </p>
    </div>
  </div>
</div>
`;
